src/gsi-ncdiag/gsi_ncdiag.py

```diff
@@ -1395,12 +1389,12 @@ def toIODAobs(self, OutDir, ObsBias, QCVars, TestRefs, clobber=True):
         for key, value2 in chan_metadata_dict.items():
             try:
                 if value2 in chan_metadata_int:
-                    outdata[(value2, 'MetaData')] = self.var(key).astype(np.int32)
+                    outdata[(value2, 'VarMetaData')] = self.var(key).astype(np.int32)
```

Comment:
Are we positive we want `VarMetaData` in here? I thought that was a deprecated group?

Reply:
The nvars dimension and the VarMetaData group have been deprecated. These come from an attempt to organize meta data long before the OBS Conventions were developed.

All meta data variables should be placed in the MetaData group and each variable should be dimensioned by a dimension of whatever is being described. Ie, channel frequencies, channel polarization etc should be dimensioned by nchans; latitude, longitude dimensioned by nlocs, etc.

The name of the dictionary `chan_metadata_dict` seems to imply that these variables are MetaData describing channels. If that is the case please change these back to MetaData and nchans; otherwise let's discuss further and figure out what dimensions to use (we still want these variables in the MetaData group). 

Thanks!

Reply:
Thank you both! The Discover is down today. I cannot remember why I made this change. Have to check once the computer is up.

Reply:
This VarMetaData is rename back to the MetaData in a new PR https://github.com/JCSDA-internal/ioda-converters/pull/1038 . The new PR converts GNSS diagnostic variables and other data in GSI .